### PR TITLE
MGMT-18305: Add certs to ingress when deploying in non-OCP clusters

### PR DESCRIPTION
**Pull request by carbonin**
Resolves https://issues.redhat.com/browse/MGMT-18305

## List all the issues related to this PR

- [ ] New Feature 
- [x] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [x] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [x] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [x] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/controller/controllers/kube_compat.go [hunk 1]
```diff
@@ -39,6 +40,10 @@ func crdExists(ctx context.Context, c client.Client, crdName string) (bool, erro
 	return err == nil, client.IgnoreNotFound(err)
 }
 
+func ingressTLSSecretName(ingressName string) string {
+	return fmt.Sprintf("%s-ingress", ingressName)
```
Comment (rccrdpccl) [inline internal/controller/controllers/kube_compat.go:44]: Where is this naming convention coming from? Not very important, but if we have control over id, should we add something to the name that would make it recognizable that's about TLS specifically? i.e. `%s-ingress-tls`

**Comment by rccrdpccl** [comment]
/hold
/lgtm
/approve

Just minor comment, please unhold if we can't/won't address it right now

File: internal/controller/controllers/kube_compat.go (hunk 1 above)
Comment (carbonin) [reply internal/controller/controllers/kube_compat.go:44]: >Where is this naming convention coming from?

Just what I happened to think of.

>should we add something to the name that would make it recognizable that's about TLS specifically? i.e. %s-ingress-tls

The secrets are created with type tls so I don't think we really need to repeat it in the name. Plus I'm not sure what other ingress-related secrets we're planning to create:

```
$ kubectl -n assisted-installer get secret
assisted-image-service-ingress   kubernetes.io/tls   3      2d23h
assisted-service-ingress         kubernetes.io/tls   3      2d23h
...
```

File: internal/controller/controllers/agentserviceconfig_controller.go [hunk 2]
```diff
@@ -1195,16 +1194,18 @@ func newAssistedCM(ctx context.Context, log logrus.FieldLogger, asc ASC) (client
...
 			"DATA_UPLOAD_ENDPOINT":   "https://console.redhat.com/api/ingress/v1/upload",
 
-			"NAMESPACE":       asc.namespace,
-			"INSTALL_INVOKER": "assisted-installer-operator",
+			"NAMESPACE":              asc.namespace,
+			"INSTALL_INVOKER":        "assisted-installer-operator",
+			"SKIP_CERT_VERIFICATION": "False",
 		}
-		// enable https only on OCP
+		// serve https only on OCP
```
Comment (CrystalChun) [inline internal/controller/controllers/agentserviceconfig_controller.go:1201]: Sorry, for my understanding - is this comment still true if we're always enabling https here https://github.com/openshift/assisted-service/pull/6564/files#diff-c619bb36181b5e1dd76fb1fcc1f54068b16a8761f77106052fcbf5e2f5f9d356R1064

Comment (carbonin) [reply internal/controller/controllers/agentserviceconfig_controller.go:1201]: Yeah, so the ingress will be https, but tls will terminate there. Meaning assisted-service will serve http between itself and the ingress controller.

That's why I changed this comment to say that we're enabling https in that the agent will expect https and we should provide the agent with certs to trust the ingress, but the assisted service itself is not serving https.

**Review by CrystalChun — APPROVED** [review]
/lgtm

File: internal/controller/controllers/agentserviceconfig_controller.go (hunk 2 above)
Comment (CrystalChun) [reply internal/controller/controllers/agentserviceconfig_controller.go:1201]: Thanks for the explanation!

**Pull request merged**